db/src/lib.rs

```diff
@@ -35,6 +32,34 @@ pub async fn create_user(pool: &PgPool, user_name: &str) -> color_eyre::Result<U
     Ok(result.user_id)
 }
 
+pub async fn add_url(pool: &PgPool, url: &str, allow_existing: &bool) -> color_eyre::Result<String> {
+    let exist_record = sqlx::query_as!(
+        ExistRecord,
+        "SELECT EXISTS(SELECT 1 FROM Page WHERE url = $1) as exists",
```

Comment:
This totally works!


I _might_ have written this as something more like 
```
SELECT page_id from Pages WHERE url = $1
```

and then use `fetch_optional` to get the result back as an option. And then `Some(_)` would mean the record exists and `None` would mean its missing!

But I don't see a real reason to change this! Just chatting!

Reply:
Love the suggestion